File: 8_kyu/Convert_number_to_reversed_array_of_digits.lua
-- Converts number n into the decimal digits.
-- Returns the digits in an array, least significant digit at index 1.
function digitize(n)
  if n == 0 then
    return {0}
  end
  local digits = {}
  while n > 0 do
    table.insert(digits, n % 10)
    n = n // 10
  end
  return digits
end

return digitize

File: 8_kyu/Convert_number_to_reversed_array_of_digits.py
def digitize(n: int) -> list[int]:
    """
    Converts number to reversed array of digits.
    """
    if n < 0 or not isinstance(n,int):
        raise ValueError("only non-negative int possible")
    if n == 0:
        return [0]
    accu = []
    while n > 0:
        accu.append( n % 10 )
        n //= 10
    return accu
